src/java.base/share/classes/jdk/internal/util/ImmutableBitSetPredicate.java

```diff
@@ -79,7 +80,35 @@ private static int wordIndex(int bitIndex) {
      * @since 22
      */
     public static IntPredicate of(BitSet original) {
+        if (original.size() <= 128) {
+            return new SmallImmutableBitSetPredicate(original);
+        }
         return new ImmutableBitSetPredicate(original);
     }
 
+    public static class SmallImmutableBitSetPredicate implements IntPredicate {
+        private final long first;
+        private final long second;
+        private SmallImmutableBitSetPredicate(BitSet original) {
+            // If this class is made public, we need to do
+            // a defensive array copy as certain BitSet implementations
```

Comment:
Outdated comments about defensive copy here, as we already copy the longs by value

Reply:
Removed.